queries.sql

```diff
@@ -8,3 +8,105 @@ SELECT name, escape_attempts From animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = TRUE;
 SELECT * FROM animals WHERE name != 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+/**** Query and update animals table ****/
+
+/* Inside a transaction update the animals table by setting the species column to unspecified.
+Verify that change was made. Then roll back the change and verify that the species 
+columns went back to the state before the transaction. * */
+BEGIN TRANSACTION;
+SAVEPOINT SP1;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK TO SP1;
+
+/*
+-Inside a transaction:
+-Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+-Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+-Commit the transaction.
+-Verify that change was made and persists after commit. 
+*/
+BEGIN TRANSACTION;
+UPDATE animals 
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+COMMIT; 
+
+/*
+-Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
+-After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)
+*/
+
+BEGIN TRANSACTION;
+DELETE FROM animals;
+ROLLBACK;
+
+/* 
+-Inside a transaction:
+-Delete all animals born after Jan 1st, 2022.
+-Create a savepoint for the transaction.
+-Update all animals' weight to be their weight multiplied by -1.
+-Rollback to the savepoint
+-Update all animals' weights that are negative to be their weight multiplied by -1.
+-Commit transaction
+*/
+
+BEGIN TRANSACTION;
+DELETE FROM animals
+WHERE date_of_birth > '2022-01-01';
+
+SAVEPOINT SP2;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+
+ROLLBACK TO SP2;
+
+UPDATE animals
+
+SET  weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+
+COMMIT; 
+
+/*********************************************/
+
+/* Write queries to answer the following questions: */ 
+
+/* How many animals are there? */
+
+SELECT COUNT(name) FROM animals;
+
+/* How many animals have never tried to escape? */
+
+SELECT COUNT(name) FROM animals
+WHERE animals.escape_attempts = 0;
+
+/* What is the average weight of animals? */
+
+SELECT AVG(weight_kg) FROM animals;
+
+/* Who escapes the most, neutered or not neutered animals? */
+
+SELECT neutered, SUM(escape_attempts) as total_escape_attempts
+FROM animals
+GROUP BY neutered;
```

Comment:
- [ ] Please you can better write the query for who escapes the most, as seen below
**Take a Look**
```sql
 SELECT neutered, AVG(escape_attempts) FROM animals
   GROUP BY neutered;
```
Or

```sql
SELECT neutered, MAX(escape_attempts) FROM animals
  GROUP BY neutered;
```